pkg/component/runtime/runtime.go

```diff
@@ -160,15 +159,11 @@ func (s *componentRuntimeState) runLoop() {
 }
 
 func (s *componentRuntimeState) getCurrent() component.Component {
-	s.currCompMx.RLock()
-	defer s.currCompMx.RUnlock()
-	return s.currComp
+	return *s.currComp.Load()
```

Comment:
This will break when the atomic pointer holds a nil value.
I guess we need a nil check here and some zero value to be returned instead

Reply:
Please look [up](https://github.com/elastic/elastic-agent/pull/7369/files#diff-25a0cbe41ced249892ab5090a7a2dacc81349647ea4d99361df6c0d5fb0079c0R119). The invariant that is held is that this always holds a non-nil value.